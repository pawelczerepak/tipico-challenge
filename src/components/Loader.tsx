import PulseLoader from 'react-spinners/PulseLoader';
import React from 'react';
import styled from 'styled-components';

export const Loader = () => {
    return <PulseLoader color="#c8102e"/>;
};

export const CenteredLoaderWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`;
