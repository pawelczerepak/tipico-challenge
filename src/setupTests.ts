import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axiosMiddleware from 'redux-axios-middleware';
import client from './apiClient';

configure({ adapter: new Adapter() });

export const mockStore = configureMockStore([thunk, axiosMiddleware(client)]);
