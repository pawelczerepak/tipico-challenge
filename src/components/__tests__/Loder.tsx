import React from 'react';
import '../../setupTests';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {Loader} from '../Loader';

describe('Loader', () => {
    it('capture snapshot', () => {
        expect(toJson(shallow(
            <Loader/>,
        )))
            .toMatchSnapshot();
    });
});
