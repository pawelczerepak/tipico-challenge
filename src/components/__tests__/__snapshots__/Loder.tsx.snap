// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader capture snapshot 1`] = `
<Loader
  color="#c8102e"
  css={Object {}}
  loading={true}
  margin="2px"
  size={15}
  sizeUnit="px"
/>
`;
